### PR TITLE
Creates New Homepage

### PR DESCRIPTION
**Pull request by surabhigupta2029**
image

image

### CODE REVIEW

File: src/Pages/Home/Home.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import React, { Component } from 'react';
```
Comment (evanugarte) [inline src/Pages/Home/Home.js:1]: remove commented out code in this file if you can

**Review by evanugarte — COMMENTED** [review]
Instead of committing files directly to core v4 can we make use of s3/some file hosting service? it makes the project much smaller. i wrote about this in #164

File: api/cloud_api/server.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ const { model } = require('mongoose');
 const { SceHttpServer } = require('../util/SceHttpServer');
 
 function main() {
-  const API_ENDPOINTS =  __dirname + '/routes/';
+  const API_ENDPOINTS = __dirname + '/routes/';
```
Comment (IsitaB) [inline api/cloud_api/server.js:5]: hm

File: api/util/SceHttpServer.js [hunk 3]
```diff
@@ -86,7 +86,7 @@ class SceHttpServer {
         useUnifiedTopology: true,
         useCreateIndex: true,
       })
-      .then(() => {})
+      .then(() => { })
```
Comment (IsitaB) [inline api/util/SceHttpServer.js:89]: dont tell me you use prettier

File: nginx.conf [hunk 4]
```diff
@@ -19,9 +19,6 @@ http {
     upstream sce-logging-api {
```
Comment (IsitaB) [inline nginx.conf:19]: take this file out of your PR

please stop using `git add .`

File: src/Components/Footer/footer.css [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
+  font-family: 'Montserrat';
 }
 
 .footer-text a {
   color: #E1F5FE;
   text-decoration: none;
+  font-family: 'Montserrat';
+}
+
+b{
```
Comment (IsitaB) [inline src/Components/Footer/footer.css:19]: format this file please

File: src/Components/Navbar/navbar.css [hunk 6]
```diff
@@ -6,6 +6,7 @@
   top: 0px;
   position: fixed;
   z-index: 4;
+  position: fixed;
```
Comment (IsitaB) [inline src/Components/Navbar/navbar.css:9]: thonk

File: src/Pages/Home/Home.js [hunk 7]
```diff
@@ -1,61 +1,271 @@
...
+  { title: 'Join our public accounts on Discord, Slack,' },
+  { title: 'Instagram and more to connect with SCE members!' },
+];
+
+// The sub block texts
+// ( the part with the empty clubroom in the background )
+const subBlockOneText = [
+  { title: 'As an SCE member, you will get inside scoops ' },
+  { title: '& opportunities.' },
+  { title: <br /> },
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:31]: 👀

File: src/Pages/Home/Home.js [hunk 8]
```diff
@@ -1,61 +1,271 @@
...
+
+const blockTwoText = [
+  { title: 'Join our public accounts on Discord, Slack,' },
+  { title: 'Instagram and more to connect with SCE members!' },
+];
+
+// The sub block texts
+// ( the part with the empty clubroom in the background )
+const subBlockOneText = [
+  { title: 'As an SCE member, you will get inside scoops ' },
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:29]: is there a better way of doing this than multiple lines? How about a block that wraps the text? Is this an animation thing?

The background for this block turns into a grey gradient when the window is too small to display the background image. Was this intentional? Is there a better way to handle this?

File: src/Pages/Home/Home.js [hunk 9]
```diff
@@ -1,61 +1,271 @@
...
-        height="850" frameborder="0" scrolling="no"/>
-        */
+import { Row, Col } from 'reactstrap';
+
+const SCEtitle = 'Software and \n Computer \n Engineering \n Society';
+
+// The major block texts surrounding the sub block
+const blockOneText = [
+  { title: 'Our clubroom offers everything an SE ' },
+  { title: 'or COMPE student could need.' },
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:14]: for consistency, if we're sticking to acronyms, could we do SE and CMPE

Could we skip abbreviations altogether?

File: src/Pages/Home/Home.js [hunk 10]
```diff
@@ -1,61 +1,271 @@
...
-        <Iframe url="https://calendar.google.com/calendar/b/4/embed?showTitle=
-        0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;showTz=0&amp;
-        height=600&amp;wkst=1&amp;bgcolor=%23ffffff&amp;src=llv828585faitko1
-        m2nh39s3js%40group.calendar.google.com&amp;color=%23182C57&amp;ctz=
-        America%2FLos_Angeles" style="border-width:0" width="1000"
-        height="850" frameborder="0" scrolling="no"/>
-        */
+import { Row, Col } from 'reactstrap';
+
+const SCEtitle = 'Software and \n Computer \n Engineering \n Society';
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:9]: I like the title better when it was a bit larger. 

Is this `vw` and `vh`'d?

The resizing is a little doomed

File: src/Pages/Home/Home.js [hunk 11]
```diff
@@ -1,61 +1,271 @@
...
+
+const subBlockThreeText = [
+  { title: 'Join us for fun and exciting social events for different' },
+  { title: 'occasions!' },
+  { title: <br /> },
+  { title: 'We host potlucks, movie nights, game nights and much more!' },
+];
+
+// Club social media icon links
+const icons = [
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:52]: hm could we move these icons to another constants file

File: src/Pages/Home/Home.js [hunk 12]
```diff
@@ -1,61 +1,271 @@
...
+      '845c-.796 0-1.441.645-1.441 1.44s.645 1.44 1.441 1.44c.795 0 1.',
+      '439-.645 1.439-1.44s-.644-1.44-1.439-1.44z'
+    ].join('')
+  }
+];
+
 
 class Home extends Component {
   render() {
+    AOS.init();
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:113]: oo

File: src/Pages/Home/Home.js [hunk 13]
```diff
@@ -1,61 +1,271 @@
...
-                // eslint-disable-next-line
-                url='https://calendar.google.com/calendar/b/4/embed?showTitle=0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;showTz=0&amp;height=600&amp;wkst=1&amp;bgcolor=%23ffffff&amp;src=llv828585faitko1m2nh39s3js%40group.calendar.google.com&amp;color=%23182C57&amp;ctz=America%2FLos_Angeles'
-                className='calendar'
-                allowFullScreen
-              />
-            </div>
+          <div class="sceLogo" data-aos="fade-down" data-aos-duration="1000" >
+            <img class="logo-styles"
+              src='images/sce_logo_2.svg'
+              alt="sce logo" style={{ width: '35%', float: 'right' }}></img>
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:126]: ma'am is this inline styling?

File: src/Pages/Home/Home.js [hunk 14]
```diff
@@ -1,61 +1,271 @@
...
+                      );
+                    })}
+                  </div>
+                </Col>
+              </Row>
+            </div>
+
+            <div className="block-space"></div>
+
+            {/* The sub block section */}
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:171]: could you make both white divs shorter?

There's a lot of empty space

Comment (IsitaB) [reply src/Pages/Home/Home.js:171]: Also that isnt even the club room but ig you put it in the other div so jank

File: src/Pages/Home/Home.js [hunk 15]
```diff
@@ -1,61 +1,271 @@
...
+                    data-aos-duration="1000" >
+                    <img className="block-images"
+                      src='images/Tesla2.jpg'
+                      alt="blockOneImg"
+                      style={{ width: '45vw', marginRight: '15%' }}></img>
+                  </div>
+                </Col>
+              </Row>
+            </div>
+            <div className="block-space"></div>
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:265]: you could've used a self closing tag 😔

File: src/Pages/Home/Home.js [hunk 16]
```diff
@@ -1,61 +1,271 @@
...
+                    </div>
+                  </Col>
+                  <Col xs={3}>
+                    <div class="subBlock-title" data-aos="fade-right"
+                      data-aos-duration="1000" >
+                      Social Events
+                    </div>
+                    <div class="subBlockContainer" data-aos="fade-right"
+                      data-aos-duration="2000" >
+                      {subBlockThreeText.map((title) => {
```
Comment (IsitaB) [inline src/Pages/Home/Home.js:210]: interesting

**Review by IsitaB — CHANGES_REQUESTED** [review]
Absolutely gorgeous!

Stylistically, I'd say make the white divs shorter. The text and logo a hint bigger.

The console has a bunch of errors and the resizing is a bit doomed
Screen Shot 2021-02-08 at 6 36 04 PM

**Review by IsitaB — APPROVED** [review]
good stuff

**Review by aashrav — APPROVED** [review]
Looks great....only thing that you should change is the image name for "1.png". Try changing it to something more descriptive/meaningful

**Review by seemavora — APPROVED** [review]
Looks much more professional, (no more slideshow woo woo)the animations are added in the perfect places, I honestly have no suggestions. This looks professional, and also comforting. Great work!!!!

**Review by nanarboursal — APPROVED** [review]
The homepage looks so so so good!!!!

File: src/Pages/Home/Home.js [hunk 17]
```diff
@@ -1,61 +1,187 @@
...
             </div>
           </div>
-          <Footer />
-        </div>
+          <div className="sceTitle" data-aos="fade-down"
+            data-aos-duration="1000" >
+            {SCEtitle}
+          </div>
+
+          <section id='things'>
```
Comment (charlotte-zhuang) [inline src/Pages/Home/Home.js:69]: Can we rename to `id='content'`?

File: src/Pages/Home/home.css [hunk 18]
```diff
@@ -1,20 +1,222 @@
...
+  /* style={{ width: '50%', float: 'right' }}*/
+}
+
+.logo-styles{
+  width:20vw;
+  height:20vw;
+}
+
+.contentContainer{
+  width: auto;
```
Comment (charlotte-zhuang) [inline src/Pages/Home/home.css:74]: `overflow-x: hidden` fixes the horizontal scrolling issue

File: src/Pages/Home/home.css [hunk 19]
```diff
@@ -1,20 +1,222 @@
...
+  background-repeat: no-repeat;
+  height: 100%;
+}
+
+.subBlock-title{
+  scroll-behavior: [ smooth ];
+  color: white;
+  position: relative;
+  font-family: 'Montserrat';
+  font-size: 1.75vw;
```
Comment (charlotte-zhuang) [inline src/Pages/Home/home.css:107]: Resizing using viewport width doesn't work well.

**Review by charlotte-zhuang — COMMENTED** [review]
Great work!

There are some CSS bugs that I noticed:

### Overflow causing horizontal scrolling

The horizontal scroll issue is still present. It looks like either some of the animation containers are the culprit, or because width is being calculated to fill the viewport width, which includes the vertical scrollbar.

I think an acceptable solution is to add `overflow-x: hidden` to the `contentContainer` div inside the `things` section. Also, can we rename the section to something better like `id="content"`?

Screen Shot 2021-02-15 at 9 32 24 PM

### Font size and spacing

Resizing the page is causing the font of the animated text content to resize in undesirable ways. Unless the page is almost fullscreen on my laptop, spacing becomes too much or too little depending on the exact dimensions of my window. Font size also becomes too small sometimes.

The rows sometimes are too short and can't display the full content of the text, but that might not be an easy fix.

Screen Shot 2021-02-15 at 9 33 28 PM

**Review by charlotte-zhuang — COMMENTED** [review]
The fixes look good, there are two issues that I noticed. Maybe someone else could see if they're small enough to ignore?

1. Text still displays poorly at certain window dimensions. Unreadable on mobile screens.
2. Animation doesn't work for rows 1 and 2 until you resize the window.

Screen Shot 2021-02-16 at 1 22 25 PM

Screen Shot 2021-02-16 at 1 23 24 PM

**Review by IsitaB — COMMENTED** [review]
Dont render the title block and logo on mobile; it covers the officers picture rn

**Comment by surabhigupta2029** [comment]
> Dont render the title block and logo on [mobile](https://enhanced-links.com/s/?q=mobile); it covers the officers picture rn

**Comment by surabhigupta2029** [comment]
Mobile screens
image
image

**Review by IsitaB — COMMENTED** [review]
Screen Shot 2021-03-03 at 1 22 30 PM
please add padding on the sides
can you make the text vertically centered

**Comment by IsitaB** [comment]
> Screen Shot 2021-03-03 at 1 22 30 PM
> 
> please add padding on the sides can you make the text vertically centered

rebase

File: src/Pages/EventManager/EventManagerModal.js [hunk 20]
```diff
@@ -61,8 +61,8 @@ function EventManagerModal(props) {
         defaultValue: props.title,
         placeholder: 'e.g. Python Workshop',
         handleChange: e => setTitle(e.target.value),
-        ifRequirementsNotMet: (title == undefined || title == '')
-         && clickedSubmit && (
+        ifRequirementsNotMet: (title === undefined || title === '')
```
Comment (evanugarte) [inline src/Pages/EventManager/EventManagerModal.js:64]:
```suggestion
        ifRequirementsNotMet: (!title)
```

**Review by evanugarte — CHANGES_REQUESTED** [review]
Be sure to remove any unused photos

File: src/Pages/Home/home.css [hunk 21]
```diff
@@ -1,20 +1,235 @@
...
+  white-space: pre-line;
+  text-align: left;
+  direction: ltr;
+  font-size: 6vh;
+  font-weight: bolder !important; 
+  line-height: 1.05;
+  letter-spacing: 1px;
+  text-shadow:  0 1px 10px #777;
+  z-index: 0;
+  overflow: auto;
```
Comment (charlotte-zhuang) [inline src/Pages/Home/home.css:62]: `overflow: visible;`

File: src/Pages/Home/home.css [hunk 22]
```diff
@@ -24,3 +239,492 @@
...
+    white-space: pre-line;
+    text-align: left;
+    direction: ltr;
+    font-size: 2.5em;
+    font-weight: bolder !important; 
+    line-height: 1.05;
+    letter-spacing: 1px;
+    text-shadow:  0 1px 10px #777;
+    z-index: 0;
+    overflow: auto;
```
Comment (charlotte-zhuang) [inline src/Pages/Home/home.css:648]: `overflow: visible;`

File: src/Pages/Home/home.css [hunk 23]
```diff
@@ -24,3 +239,492 @@
...
+    white-space: pre-line;
+    text-align: left;
+    direction: ltr;
+    font-size: 4.5vw;
+    font-weight: bolder !important; 
+    line-height: 1.05;
+    letter-spacing: 1px;
+    text-shadow:  0 1px 10px #777;
+    z-index: 0;
+    overflow: auto;
```
Comment (charlotte-zhuang) [inline src/Pages/Home/home.css:505]: `overflow: visible;`

File: src/Pages/Home/home.css [hunk 24]
```diff
@@ -24,3 +239,492 @@
...
```
Comment (charlotte-zhuang) [inline src/Pages/Home/home.css:319]: `overflow: visible;`

File: src/Pages/Home/Home.js (hunk 10 above)
Comment (charlotte-zhuang) [reply src/Pages/Home/Home.js:10]: Resizing with `vw` and `vh` isn't working well. Could we use fixed sizes for each `@media` size?

**Review by charlotte-zhuang — CHANGES_REQUESTED** [review]
## Comments

1. Resizing still has a lot of issues, especially between `@media` transitions. I agree that using `vh` and `vw` to resize content doesn't work well. Could we change to fixed sizes? (see screenshots and comments)
2. Some changes requested for mobile displays. (see screenshots and comments)
3. Animations still don't load until you resize the window.
4. The social media buttons need a different color in dark mode (you can barely see them)

## Mobile & Dark Mode

Add more space for the title. Make the row 2 text larger and give it more margins on the sides.

Screen Shot 2021-03-03 at 7 11 23 PM

Social media buttons cover text in row 3. Buttons are very hard to see in dark mode.

Screen Shot 2021-03-03 at 7 11 02 PM

## Screen Resizing

There are lots of display issues at different screen sizes.

Screen Shot 2021-03-03 at 7 12 19 PM

Screen Shot 2021-03-03 at 7 12 40 PM

**Pull request closed without merging**